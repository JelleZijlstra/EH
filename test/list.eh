#!/usr/bin/ehi

include '../lib/library.eh'

l = Cons 1, (Cons 2, (Cons 3, Nil))
echo l.toString()

echo l.length()

echo (l.map 2.operator*).toString()

echo l.reduce 1, (accum, val => accum * val)

echo (l.reverse()).toString()

l2 = Cons 4, (Cons 2, (Cons 3, (Cons 1, Nil)))

echo (l2.sort()).toString()

echo (l2.append (Cons 1, Nil)).toString()
