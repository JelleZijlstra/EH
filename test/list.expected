1::2::3::[]
3
2::4::6::[]
6
3::2::1::[]
1::2::3::4::[]
4::2::3::1::1::[]
